wallet/src/ui/app/redux/slices/txresults/index.ts

```diff
@@ -0,0 +1,154 @@
+    results
+        ? results
+              .map((result) => result[1])
+              .filter((value, index, self) => self.indexOf(value) === index)
+        : [];
+
+export const getTransactionsByAddress = createAsyncThunk<
+    TxResultByAddress,
+    void,
+    AppThunkConfig
+>(
+    'sui-transactions/get-transactions-by-address',
+    async (_, { getState, extra: { api } }): Promise<TxResultByAddress> => {
+        const address = getState().account.address;
+
+        if (!address) {
+            return [];
+        }
+        // Get all transactions txId for address
+        const transactions: GetTxnDigestsResponse = (
+            await api.instance.getTransactionsForAddress(address)
+        ).filter((tx) => tx);
+
+        if (!transactions || !transactions.length) {
+            return [];
+        }
+        //getTransactionWithEffectsBatch
+        const resp = await api.instance
+            .getTransactionWithEffectsBatch(deduplicate(transactions))
+            .then((txEffs: TransactionEffectsResponse[]) => {
+                return (
+                    txEffs
+                        .map((txEff, i) => {
+                            const [seq, digest] = transactions.filter(
+                                (transactionId) =>
+                                    transactionId[1] ===
+                                    getTransactionDigest(txEff.certificate)
+                            )[0];
+                            const res: CertifiedTransaction = txEff.certificate;
+                            // TODO: handle multiple transactions
+                            const txns = getTransactions(res);
+                            if (txns.length > 1) {
+                                return null;
+                            }
+                            const txn = txns[0];
+                            const txKind = getTransactionKindName(txn);
+                            const recipient =
+                                getTransferCoinTransaction(txn)?.recipient;
+
+                            return {
+                                seq,
+                                txId: digest,
+                                status: getExecutionStatusType(txEff),
+                                txGas: getTotalGasUsed(txEff),
+                                kind: txKind,
+                                From: res.data.sender,
+                                ...(recipient
+                                    ? {
+                                          To: recipient,
+                                      }
+                                    : {}),
+                            };
+                        })
+                        // Remove failed transactions and sort by sequence number
+                        .filter((itm) => itm)
+                        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+                        .sort((a, b) => b!.seq - a!.seq)
```

Comment:
The adapter has a [sorting functionality](https://redux-toolkit.js.org/api/createEntityAdapter#sortcomparer) as well . I guess it's the same for now that we only populate it with this action